dell/xps/15-9500/xps-common.nix

```diff
@@ -0,0 +1,24 @@
+{ lib, ... }:
+let
+  thermald-conf = ./thermald-conf.xml;
+in
+{
+  imports = [
+    ../../../common/cpu/intel
+    ../../../common/pc/laptop
+    ../../../common/pc/laptop/ssd
+  ];
+
+  # Boot loader
+  boot.kernelParams = lib.mkDefault [ "acpi_rev_override" ];
+
+  # This will save you money and possibly your life!
+  services.thermald.enable = lib.mkDefault true;
+
+  # Thermald doesn't have a default config for the 9500 yet, the one in this repo
+  # was generated with dptfxtract-static (https://github.com/intel/dptfxtract)
+  services.thermald.configFile = lib.mkDefault thermald-conf;
+
+  # Set the tlp config to powersave explictly. TLP is enabled in common/pc/laptop.
+  services.tlp.extraConfig = lib.mkDefault "CPU_SCALING_GOVERNOR_ON_AC=powersave\nCPU_SCALING_GOVERNOR_ON_BAT=powersave";
```

Comment:
```suggestion
  services.tlp.extraConfig = lib.mkDefault ''
    CPU_SCALING_GOVERNOR_ON_AC=powersave
    CPU_SCALING_GOVERNOR_ON_BAT=powersave
  '';
```

Reply:
Ok. `powersave` is recommended by upstream: Should be fine than: https://linrunner.de/tlp/settings/processor.html?highlight=cpu_scaling_governor_on_ac

Reply:
It looks like tpm does not have any default cpu govenor set and our nixos-config-generate script sets `powersave` by default. I wonder where `performance` is set?

Reply:
I just found out that if tlp is enabled the generated cpu scaling govenor is overwritten:  https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/services/hardware/tlp.nix#L55

Reply:
I am currently preparing a fix to fix this in our tlp module instead.

Reply:
Sounds good!

Reply:
Here is the fix: https://github.com/NixOS/nixpkgs/pull/94064